pkg/processor/procbuiltin/extractfield_test.go

```diff
@@ -206,13 +206,19 @@ func TestExtractFieldPayload_Process(t *testing.T) {
 			Settings: map[string]string{extractFieldConfigField: "foo"},
 		},
 		args: args{r: record.Record{
-			Payload: record.StructuredData{
-				"foo": 123,
+			Payload: record.Change{
+				Before: nil,
+				After: record.StructuredData{
+					"foo": 123,
+				},
```

Comment:
IIRC, the decision for this and similar built-in processors was to operate on the `after` field. If so, it would be good to mention that in the docs.

Also, let's say we use extractfieldpayload. Should the output be a Change, where `before` remains the same, and `after` is transformed, or should be just some raw data with the extracted field?

Reply:
Exactly, all built-in transforms were changed to use `Payload.After` instead of just `Payload`. I'll add it to the documentation, I just have to find the right place for this info 🙂

I'm not sure I understand the question about `extractfieldpayload` - the output is the same as before, only that instead of manipulating `Payload` it's manipulating `Payload.After`.

Reply:
I updated the processor comments and the processor doc.

Reply:
> I'm not sure I understand the question about `extractfieldpayload` - the output is the same as before, only that instead of manipulating `Payload` it's manipulating `Payload.After`.

Nevermind, my question is actually not valid.:) Prior to this change, record.Payload was being replaced with the field.

So I was wondering if we are supposed to do that here too (and not replace record.Payload.After with the extracted field). But that actually makes no sense, since record.Payload is a `Change`, not `Data`.